### ISSUE TITLE
whitespace 

### ISSUE THREAD

**Issue by okomarov** [matlab2tikz/matlab2tikz#808]
Can we decide for one standard so that I can develop a small tool to manually run before issuing a PR?

I suggest for Matlab's smart indent + remove whitespace on empty lines. Hence, without the 2space indentation in a function.

What are your thoughts?

**Comment by egeerardyn** [comment]
Sure!

My personal preference is:
 - no trailing spaces if possible
 - 4 spaces (not tabs) per level: 2 spaces is often not enough
 - [Indent all functions](http://blogs.mathworks.com/community/2009/05/11/keep-your-code-readable-with-smart-indenting/)

However, I'm happy with any agreement (only spaces vs. tabs I'm completely in the spaces camp).

However, the problem with automatic formatters is that these often ruin properly crafted code, e.g. line continuations:
```matlab
% crafted
var = struct('bla', 'bla', ...
             'foo', 'foo', ...
             'baz', 'baz');
```
versus
```matlab
% smart-ident
var = struct('bla', 'bla', ...
    'foo', 'foo', ...
    'baz', 'baz');
```

Luckily, smart-indent leaves space in the middle intact (although we could do with some guidelines there as well: spaces after each comma).

In any case, if we can formulate some rules, I'm quite convinced there might be some plugins for Jenkins available to check those rules such that in the future we don't pull oddly-formatted code.

**Comment by egeerardyn** [comment]
I will leave this open as a reminder to myself to fix indentation before the 1.1 release of m2t.
